src/main/java/barcus/parser/Parser.java

```diff
@@ -0,0 +1,123 @@
+                    //int pos = Integer.parseInt(words[1]);
+                    return new MarkCommand(pos);
+                } catch (NumberFormatException e) {
+                    throw new BarcusException("please have an integer after 'mark'");
+                }
+            }
+        } else if (words[0].equals("todo")) {
+            if (words.length < 2) {
+                throw new BarcusException("please include a description of the todo");
+            } else {
+                return new AddTodoCommand(String.join(" ", Arrays.copyOfRange(words, 1, words.length)));
+            }
+        } else if (words[0].equals("deadline")) {
+            List<String> wordsList = Arrays.asList(words);
+            if (!wordsList.contains("/by")) {
+                throw new BarcusException("please include '/by' and deadline after it");
+            } else {
+                int byI = wordsList.indexOf("/by");
+                return new AddDeadlineCommand(
+                        String.join(" ", Arrays.copyOfRange(words, 1, byI)),
+                        String.join(" ", Arrays.copyOfRange(words, byI + 1, words.length)));
+            }
+        } else if (words[0].equals("event")) {
+            List<String> wordsList = Arrays.asList(words);
+            if (!wordsList.contains("/from") || !wordsList.contains("/to")) {
+                throw new BarcusException("please include '/from' and '/to' "
+                        + "as well as dates after each of those words");
+            } else {
+                int fromI = wordsList.indexOf("/from");
+                int toI = wordsList.indexOf("/to");
+                return new AddEventCommand(
+                        String.join(" ", Arrays.copyOfRange(words, 1, fromI)),
+                        String.join(" ", Arrays.copyOfRange(words, fromI + 1, toI)),
+                        String.join(" ", Arrays.copyOfRange(words, toI + 1, words.length)));
+            }
+        } else if (words[0].equals("delete")) {
+            if (words.length == 1) {
+                throw new BarcusException("please have an integer after 'delete'");
+            } else {
+                try {
+                    int[] pos = new int[words.length - 1];
+                    for (int i = 1; i < words.length; i++) {
+                        pos[i - 1] = Integer.parseInt(words[i]);
+                    }
+                    return new DeleteCommand(pos);
+                } catch (NumberFormatException e) {
+                    throw new BarcusException("please have an integer after 'delete'");
+                }
+            }
+        } else if (words[0].equals("find")) {
+            if (words.length == 1) {
+                throw new BarcusException("please include what word(s) you want to find after 'find'");
+            } else {
+                return new FindCommand(String.join(" ", Arrays.copyOfRange(words, 1, words.length)));
+            }
+        }
+
+        return new UnknownCommand();
+    }
```

Comment:
Could use enum and switch statements to make the code look cleaner? It looks abit harder to read for now.
Do you think you can switch up the logic for checking if the command is valid? for example taking out the words.length check to check for multiple commands at once instead of 1 by 1